pkg/kor/replicaset.go

```diff
@@ -0,0 +1,85 @@
+package kor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"os"
+)
+
+func ProcessNamespaceReplicaSets(clientset kubernetes.Interface, namespace string, filterOpts *FilterOptions) ([]string, error) {
+	replicaSetList, err := clientset.AppsV1().ReplicaSets(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var unusedReplicaSetNames []string
+
+	for _, replicaSet := range replicaSetList.Items {
+		if replicaSet.Labels["kor/used"] == "true" {
+			continue
+		}
+
+		// checks if the resource has any labels that match the excluded selector specified in opts.ExcludeLabels.
+		// If it does, the resource is skipped.
+		if excluded, _ := HasExcludedLabel(replicaSet.Labels, filterOpts.ExcludeLabels); excluded {
+			continue
+		}
+		// checks if the resource's age (measured from its last modified time) matches the included criteria
+		// specified by the filter options.
+		if included, _ := HasIncludedAge(replicaSet.CreationTimestamp, filterOpts); !included {
+			continue
+		}
+
+		// if the replicaSet is specified 0 replica and current available & ready & fullyLabeled replica count is all 0, think the replicaSet is completed
+		if *replicaSet.Spec.Replicas == 0 && replicaSet.Status.AvailableReplicas == 0 && replicaSet.Status.ReadyReplicas == 0 && replicaSet.Status.FullyLabeledReplicas == 0 {
```

Comment:
What is `FullyLabeledReplicas`?

Reply:
fullyLabeledReplicas is the number of pods that have labels matching the labels of the pod template of the replicaSet. if a replicaSet desire to scale pods to 0, when  it complete it's work, available & ready & fullyLabeled should all be 0, and barely grow again.

Reply:
I see some unused replicasets do not contain all these fields. This will likely cause an error

Reply:
yes, i see this too, so i check all this property, that ensure do not make wrong deletion

Reply:
@cindyliu-tec But what will happen when we run this check and the field doesn't exist?
Kor will likely crash

Reply:
as golang language design, a struct field  without assignment will default assigned the zero value of it's type. so if  the field doesn't exist, it's value would be 0. when a replicaSet is still active, it must has those fields, and impossible be all 0.

Reply:
as golang language design, a struct field  without assignment will default assigned the zero value of it's type. so if  the field doesn't exist, it's value would be 0. when a replicaSet is still active, it must has those fields, and impossible be all 0.